test/command_line/test_generate_mask.py

```diff
@@ -6,89 +6,108 @@
 from dials.command_line.generate_mask import generate_mask, phil_scope
 from dials.command_line.dials_import import Script as ImportScript
 from dials.command_line.dials_import import phil_scope as import_phil_scope
-from dxtbx.serialize import load
+from dxtbx.model import ExperimentList
 from libtbx import phil
 
 
 @pytest.fixture(
     params=[
-        "centroid_test_data",
-        pytest.param(
-            "l_cysteine_dials_output",
-            marks=pytest.mark.skip(
-                reason="test depends on https://github.com/dials/data/pull/40"
-            ),
-        ),
+        {
+            "directory": "centroid_test_data",
+            "filename": "imported_experiments.json",
+            "masks": ["pixels.mask"],
+        },
+        {
+            "directory": "l_cysteine_dials_output",
+            "filename": "imported.expt",
+            "masks": ["pixels_%d.mask" % (i + 1) for i in range(4)],
+        },
     ],
     ids=["One sequence", "Four sequences"],
 )
-def input_experiment_list(request, dials_data):
-    filename = (dials_data(request.param) / "imported_experiments.json").strpath
-    return load.experiment_list(filename)
+def experiments_masks(request, dials_data):
+    filename = (
+        dials_data(request.param["directory"]) / request.param["filename"]
+    ).strpath
+    return ExperimentList.from_file(filename), request.param["masks"]
 
 
-def test_generate_mask(input_experiment_list, tmpdir):
+def test_generate_mask(experiments_masks, tmpdir):
+    experiments, masks = experiments_masks
+
     params = phil_scope.fetch().extract()
     with tmpdir.as_cwd():
-        generate_mask(input_experiment_list, params)
+        generate_mask(experiments, params)
+
+    assert all(tmpdir.join(mask).check() for mask in masks)
 
-    assert tmpdir.join("pixels.mask").check()
 
+def test_generate_mask_with_untrusted_rectangle(experiments_masks, tmpdir):
+    experiments, masks = experiments_masks
 
-def test_generate_mask_with_untrusted_rectangle(input_experiment_list, tmpdir):
     params = phil_scope.fetch(
         phil.parse("untrusted.rectangle=100,200,100,200")
     ).extract()
     params.output.experiments = "masked.expt"
     with tmpdir.as_cwd():
-        generate_mask(input_experiment_list, params)
+        generate_mask(experiments, params)
 
-    assert tmpdir.join("pixels.mask").check()
+    assert all(tmpdir.join(mask).check() for mask in masks)
     assert tmpdir.join("masked.expt").check()
 
-    experiments = load.experiment_list(tmpdir.join("masked.expt").strpath)
+    experiments = ExperimentList.from_file(tmpdir.join("masked.expt").strpath)
```

Comment:
this really, really, really makes me want to extend the support of pathlib-like objects in dxtbx/dials/xia2 😄

Reply:
Yes, I know!